### PR TITLE
Set a transaction isolation level for MySQL connections

### PR DESCRIPTION
**Pull request by timvaillancourt**
Related issue: https://github.com/github/gh-ost/issues/1155

### Description

This PR resolves https://github.com/github/gh-ost/issues/1155 by defining a connection-level transaction isolation level when setting up new MySQL connections. I also moved connection params to a slice that's joined into a string later-on because the giant params string was getting pretty unreadable. I also moved cases of `log.Fatalf` with no tokens to `log.Fatal` 🧹

At this time `REPEATABLE_READ` is used as the isolation for all connections following a discussion here: https://github.com/github/gh-ost/issues/1155

> In case this PR introduced Go code changes:

- [x] contributed code is using same conventions as original code
- [x] `script/cibuild` returns with no formatting errors, build errors or unit test errors.

### CODE REVIEW

**Comment by timvaillancourt** [comment]
Hmm works on my laptop, fails in CI. More soon! 🤔

File: go/mysql/connection.go [hunk 1]
```diff
@@ -112,12 +114,23 @@ func (this *ConnectionConfig) GetDBUri(databaseName string) string {
...
 	if this.tlsConfig != nil {
 		tlsOption = TLS_CONFIG_KEY
 	}
-	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?timeout=%fs&readTimeout=%fs&writeTimeout=%fs&interpolateParams=%t&autocommit=true&charset=utf8mb4,utf8,latin1&tls=%s", this.User, this.Password, hostname, this.Key.Port, databaseName, this.Timeout, this.Timeout, this.Timeout, interpolateParams, tlsOption)
+	connectionParams := []string{
+		"autocommit=true",
+		"charset=utf8mb4,utf8,latin1",
+		"interpolateParams=true",
+		fmt.Sprintf("tls=%s", tlsOption),
+		fmt.Sprintf("transaction_isolation=%q", transactionIsolation),
```
Comment (dm-2) [inline go/mysql/connection.go:129]: Nice!

File: go/mysql/connection.go [hunk 2]
```diff
@@ -112,12 +114,23 @@ func (this *ConnectionConfig) GetDBUri(databaseName string) string {
...
+		fmt.Sprintf("timeout=%fs", this.Timeout),
+		fmt.Sprintf("readTimeout=%fs", this.Timeout),
+		fmt.Sprintf("writeTimeout=%fs", this.Timeout),
+	}
+
+	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?%s", this.User, this.Password, hostname, this.Key.Port, databaseName, strings.Join(connectionParams, "&"))
```
Comment (dm-2) [inline go/mysql/connection.go:135]: The new formatting is so much easier to read! :+1:

**Review by dm-2 — APPROVED** [review]
LGTM :rocket:

File: go/mysql/connection.go (hunk 2 above)
Comment (timvaillancourt) [reply go/mysql/connection.go:135]: Thanks @dm-2! Yeah `gh-ost` is too hard to read 😄. Hoping to invest some time into making some things more readable 👍

I think breaking up some of the multi-page-long functions might be the next thing I look at 🤔

**Pull request merged**